components/automate-cli/cmd/chef-automate/ha_node_delete_aws.go

```diff
@@ -0,0 +1,364 @@
+	}
+	err = modifyConfigForDeleteNodeForAWS(
+		&dna.config.ChefServer.Config.InstanceCount,
+		dna.chefServerIpList,
+	)
+	if err != nil {
+		return status.Wrap(err, status.ConfigError, "Error modifying chef-server instance count")
+	}
+	err = modifyConfigForDeleteNodeForAWS(
+		&dna.config.Opensearch.Config.InstanceCount,
+		dna.opensearchIpList,
+	)
+	if err != nil {
+		return status.Wrap(err, status.ConfigError, "Error modifying opensearch instance count")
+	}
+	err = modifyConfigForDeleteNodeForAWS(
+		&dna.config.Postgresql.Config.InstanceCount,
+		dna.postgresqlIpList,
+	)
+	if err != nil {
+		return status.Wrap(err, status.ConfigError, "Error modifying postgresql instance count")
+	}
+	return nil
+}
+
+func (dna *DeleteNodeAWSImpl) prepare() error {
+	return dna.nodeUtils.taintTerraform(dna.terraformPath)
+}
+
+func (dna *DeleteNodeAWSImpl) promptUserConfirmation() (bool, error) {
+	dna.writer.Println("Existing nodes:")
+	dna.writer.Println("================================================")
+	dna.writer.Println("Automate => " + strings.Join(dna.AWSConfigIp.configAutomateIpList, ", "))
+	dna.writer.Println("Chef-Server => " + strings.Join(dna.AWSConfigIp.configChefServerIpList, ", "))
+	dna.writer.Println("OpenSearch => " + strings.Join(dna.AWSConfigIp.configOpensearchIpList, ", "))
+	dna.writer.Println("Postgresql => " + strings.Join(dna.AWSConfigIp.configPostgresqlIpList, ", "))
+	dna.writer.Println("")
+	dna.writer.Println("Nodes to be deleted:")
+	dna.writer.Println("================================================")
+	if len(dna.automateIpList) > 0 {
+		dna.writer.Println("Automate => " + strings.Join(dna.automateIpList, ", "))
+	}
+	if len(dna.chefServerIpList) > 0 {
+		dna.writer.Println("Chef-Server => " + strings.Join(dna.chefServerIpList, ", "))
+	}
+	if len(dna.opensearchIpList) > 0 {
+		dna.writer.Println("OpenSearch => " + strings.Join(dna.opensearchIpList, ", "))
+	}
+	if len(dna.postgresqlIpList) > 0 {
+		dna.writer.Println("Postgresql => " + strings.Join(dna.postgresqlIpList, ", "))
+	}
+	dna.writer.Println("Removal of nodes for Postgresql or OpenSearch is at your own risk and may result to data loss. Consult your database administrator before trying to delete Postgresql or OpenSearch nodes.")
+	return dna.writer.Confirm("This will delete the above nodes from your existing setup. It might take a while. Are you sure you want to continue?")
+}
+
+func (dna *DeleteNodeAWSImpl) runDeploy() error {
+	err := dna.runRemoveNodeFromAws()
+	if err != nil {
+		return err
+	}
+
+	err = dna.nodeUtils.moveAWSAutoTfvarsFile(dna.terraformPath)
+	if err != nil {
+		return err
+	}
+
+	err = dna.nodeUtils.modifyTfArchFile(dna.terraformPath)
+	if err != nil {
+		return err
+	}
+	err = dna.nodeUtils.writeHAConfigFiles(awsA2harbTemplate, dna.config)
+	if err != nil {
+		return err
+	}
+
+	argsdeploy := []string{"-y"}
+	err = dna.nodeUtils.executeAutomateClusterCtlCommandAsync("provision", argsdeploy, provisionInfraHelpDocs)
```

Comment:
In case of Provision Infra fails for any reason =, then we do not have flow for that